crates/subspace-networking/src/behavior/custom_record_store.rs

```diff
@@ -145,10 +146,10 @@ pub trait RecordStorage<'a> {
     fn get(&'a self, k: &Key) -> Option<Cow<'_, Record>>;
 
     /// Puts a record into the store.
-    fn put(&'a mut self, r: Record) -> store::Result<()>;
```

Comment:
Can we add `TODO` to remove this lifetime once we upgrade nightly to latest? GATs are stabilized, so we can use that here:
```rust
pub trait RecordsStorage {
    type RecordsIter<'a>: Iterator<Item = Cow<'a, Record>>;
    ...
    fn records(&'_ self) -> Self::RecordsIter<'_>;
}
```